frame/support/src/traits.rs

```diff
@@ -934,40 +934,71 @@ impl WithdrawReasons {
 	}
 }
 
-/// Trait for type that can handle incremental changes to a set of account IDs.
+/// Trait for types that can handle incremental changes to a set of account IDs.
+///
+/// Allows member changes according to the "verify first, write last" maxime by
+/// providing the `ensure_can_change_members` function for checking as well as the
+/// actual writing logic.
 pub trait ChangeMembers<AccountId: Clone + Ord> {
+
+	/// Returns whether changing the members will execute without errors.
+	///
+	/// Defaults to a noop. Implement this if `change_members_sorted` could fail.
+	///
+	/// If this returns `Err`, calls to `change_members`, `change_members_sorted` and 
+	/// `set_members_sorted` are not guaranteed to have the desired effect.
+	fn ensure_can_change_members(_new: &[AccountId], _old: &[AccountId]) -> DispatchResult {
```

Comment:
I don't have much to say agains the core of the change itself, but enforcing this trait, which by itself has _nothing to do with dispatches_, to return a `DispatchResult` is a mistake. While I can guess that it makes the code on the pallets slightly easier, I think it makes a whole lot more sense if this returns normal `Result`, or anything else which is generic, and then you can do a simple `map_err()` on your pallet. 

I guess we might use the same approach for other traits in `support` as well.. which is also wrong IMO. Either way, take it with a grain of salt, but I think it is a valid point.

Reply:
That is a good point, thanks.